### ISSUE TITLE
Shortcut methods don't implement PCancelable

### ISSUE THREAD

**Issue by xamgore** [sindresorhus/got#1417]
#### Describe the bug

 - Node.js version: v14.8
 - OS & version: macOS Catalina v10.15.2

`got.get(...).json()` is not an instance of `CancelableRequest` and `cancel()` method is not implemented:

```ts
export interface CancelableRequest<T extends Response | Response['body'] = Response['body']> extends PCancelable<T>, RequestEvents<CancelableRequest<T>> {
	json: <ReturnType>() => CancelableRequest<ReturnType>;
	buffer: () => CancelableRequest<Buffer>;
	text: () => CancelableRequest<string>;
}
```

#### Actual behavior

`got.get(...).json().cancel()` throws `TypeError { message: 'p.cancel is not a function' }`

#### Expected behavior

Cancels the `got.get(...)` promise without an error.

#### Code to reproduce

```ts
test('recover from cancellation of shortcut using error instance', async t => {
	// Canceled before connection started
	const p = got('http://example.com').buffer();
	const recover = p.catch((error: Error) => {
		if (error instanceof got.CancelError) {
			return;
		}

		throw error;
	});

	p.cancel();

	await t.notThrowsAsync(recover);
});
```

#### Checklist

- [x] I have read the documentation.
- [x] I have tried my code with the latest version of Node.js and Got.

**Comment by szmarczak** [comment]
Fixed in 1f132e88b4e9b295631637757d64307032a4e56e

**Comment by xamgore** [comment]
@szmarczak, I have decided to ensure the issue is solved with got `11.8.1` and node `v14.15.0`. And it seems the problem still holds.

Here you may see `x` is an instance of `PCancelable` and has the `.cancel()` method as expected.

<details>
<img src="https://user-images.githubusercontent.com/4586392/102539672-0ca4f300-40bf-11eb-9025-69922aae23d7.png" width="560"/>
</details>

Though, if you call `...buffer()`, the returned object will be a usual `Promise`. That comes in accordance with the `shortcut` source code, [own properties are copied](https://github.com/sindresorhus/got/blob/23695808da78b689b79febb9d5ff3a7061bf578d/source/as-promise/index.ts#L187), but `__proto__` is still `Promise` instead of `PCancelable`.

<details>
<img src="https://user-images.githubusercontent.com/4586392/102539774-2c3c1b80-40bf-11eb-96b2-210ee3b4be6e.png" width="560"/>
</details>

The test case for your convenience:

```ts
  test('cancel buffer', async () => {
    const x = client.get('https://example.com#cancel-buffer').buffer()
    x.cancel()
    await expect(() => x).rejects.toThrow(CancelError)
  })
```
